server/protocols/webadmin-cli/src/main/java/org/apache/james/cli/domain/DomainCreateCommand.java

```diff
@@ -0,0 +1,57 @@
+/******************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one     *
+ * or more contributor license agreements.  See the NOTICE file   *
+ * distributed with this work for additional information          *
+ * regarding copyright ownership.  The ASF licenses this file     *
+ * to you under the Apache License, Version 2.0 (the              *
+ * "License"); you may not use this file except in compliance     *
+ * with the License.  You may obtain a copy of the License at     *
+ *                                                                *
+ * http://www.apache.org/licenses/LICENSE-2.0                     *
+ *                                                                *
+ * Unless required by applicable law or agreed to in writing,     *
+ * software distributed under the License is distributed on an    *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY         *
+ * KIND, either express or implied.  See the License for the      *
+ * specific language governing permissions and limitations        *
+ * under the License.                                             *
+ ******************************************************************/
+
+package org.apache.james.cli.domain;
+
+import java.util.concurrent.Callable;
+
+import org.apache.james.httpclient.DomainClient;
+
+import feign.Feign;
+import feign.Response;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+    name = "create",
+    description = "Create a new domain")
+public class DomainCreateCommand implements Callable<Integer> {
+
+    @CommandLine.ParentCommand DomainCommand domainCommand;
+
+    @CommandLine.Parameters
+    String domainName;
+
+    @Override
+    public Integer call() {
+        try {
+            DomainClient domainClient = Feign.builder()
+                .target(DomainClient.class, domainCommand.webAdminCli.jamesUrl + "/domains");
+            Response rs = domainClient.createADomain(domainName);
+            if (rs.status() == 204) {
+                return 0;
+            } else {
+                return 1;
+            }
+        } catch (Exception e) {
+            e.printStackTrace(domainCommand.err);
```

Comment:
I don't think duping a stacktrace to the user is a good error reporting

Reply:
What would you do instead? Logging?